main.py

```diff
@@ -47,112 +48,145 @@ def _train_loop(
+            done = env.reb_step(model_data_pairs=model_data_pairs)
 
-            for model in models:
-                model.train_log.reward += model.actor_data.rewards.reb_reward
+            for model_data_pair in model_data_pairs:
+                model_data_pair.model.train_log.reward += (
+                    model_data_pair.actor_data.rewards.reb_reward
+                )
             # track performance over episode
-            for model in models:
-                model.actor_data.model_log.reward += model.actor_data.rewards.reb_reward
-                model.actor_data.model_log.served_demand += (
-                    model.actor_data.info.served_demand
+            for model_data_pair in model_data_pairs:
+                model_data_pair.actor_data.model_log.reward += (
+                    model_data_pair.actor_data.rewards.reb_reward
+                )
+                model_data_pair.actor_data.model_log.served_demand += (
+                    model_data_pair.actor_data.info.served_demand
                 )
-                model.actor_data.model_log.rebalancing_cost += (
-                    model.actor_data.info.rebalancing_cost
+                model_data_pair.actor_data.model_log.rebalancing_cost += (
+                    model_data_pair.actor_data.info.rebalancing_cost
                 )
-                model.rewards.append(
-                    model.actor_data.rewards.pax_reward
-                    + model.actor_data.rewards.reb_reward
+                model_data_pair.model.rewards.append(
+                    model_data_pair.actor_data.rewards.pax_reward
+                    + model_data_pair.actor_data.rewards.reb_reward
                 )
 
-                model.train_log.served_demand += model.actor_data.info.served_demand
-                model.train_log.rebalancing_cost += (
-                    model.actor_data.info.rebalancing_cost
+                model_data_pair.model.train_log.served_demand += (
+                    model_data_pair.actor_data.info.served_demand
+                )
+                model_data_pair.model.train_log.rebalancing_cost += (
+                    model_data_pair.actor_data.info.rebalancing_cost
                 )
             # stop episode if terminating conditions are met
             if done:
                 break
 
         if training:
             # perform on-policy backprop
-            for model in models:
-                model.training_step()
+            for model_data_pair in model_data_pairs:
+                model_data_pair.model.training_step()
 
         # Send current statistics to screen
         epochs.set_description(
-            f"Episode {i_episode+1} | Reward: {actor_data[0].model_log.reward:.2f} |"
-            f"ServedDemand: {actor_data[0].model_log.served_demand:.2f} "
-            f"| Reb. Cost: {actor_data[0].model_log.rebalancing_cost:.2f}"
+            f"Episode {i_episode+1} | Reward: {model_data_pairs[0].actor_data.model_log.reward:.2f} |"
```

Comment:
skal vi ikke bare outputte for begge actors? Might as well, det er bare et par ekstra linjer